contracts/Controller.sol

```diff
@@ -124,38 +130,89 @@ contract Controller is Ownable {
     /**
      * @notice Return a vault balances, depend of the short option expiry
      * @dev if vault have no short option or issued option is not expired yet, return vault, else get excess margin and return it as collateral amount inside Vault struct.
-     * @param _owner vault owner.
+     * @param _owner account owner.
      * @param _vaultId vault.
      * @return Vault struct
      */
-    // function getVaultBalances(address _owner, uint256 _vaultId) external view returns (MarginAccount.Vault memory) {
-    // }
+    function getVaultBalances(address _owner, uint256 _vaultId) external view returns (MarginAccount.Vault memory) {
+        MarginAccount.Vault memory vault = getVault(_owner, _vaultId);
+
+        if ((vault.shortOtokens.length != 0) && (isExpired(vault.shortOtokens[0]))) {
+            address calculatorModule = AddressBookInterface(addressBook).getMarginCalculator();
+            MarginCalculatorInterface calculator = MarginCalculatorInterface(calculatorModule);
+
+            (uint256 netValue, ) = calculator.getExcessMargin(vault, vault.shortOtokens[0]);
+            vault.collateralAmounts[0] = netValue;
+
+            return vault;
+        }
+
+        return vault;
+    }
 
     /**
      * @dev return if an expired oToken contract’s price has been finalized. Returns true if the contract has expired AND the oraclePrice at the expiry timestamp has been finalized.
      * @param _otoken The address of the relevant oToken.
      * @return A boolean which is true if and only if the price is finalized.
      */
-    // function isPriceFinalized(address _otoken) external view returns (bool) {
-    // }
+    function isPriceFinalized(address _otoken) external view returns (bool) {
+        address oracleModule = AddressBookInterface(addressBook).getOracle();
+        OracleInterface oracle = OracleInterface(oracleModule);
+
+        OtokenInterface otoken = OtokenInterface(_otoken);
+
+        address otokenUnderlyingAsset = otoken.underlyingAsset();
+        address otokenStrikeAsset = otoken.strikeAsset();
+        address otokenCollateralAsset = otoken.collateralAsset();
+        uint256 otokenExpiryTimestamp = otoken.expiryTimestamp();
+
+        bytes32 batch = keccak256(
+            abi.encode(otokenUnderlyingAsset, otokenStrikeAsset, otokenCollateralAsset, otokenExpiryTimestamp)
+        );
```

Comment:
use getBatch()

Reply:
yeah but still this function will need the `otoken.expiryTimestamp()`, so if I use `getBatch` will make to external call to `otoken`, so kept getBatch just for external use.

Reply:
But it looks so redundant. @aparnakr thoughts?

Reply:
yeah but two external call to `otoken` will consume more gas.

Reply:
no strong thoughts on this one. whichever is easier to understand imo.

Reply:
it's just one more external call to read expiryTimeStamp, and you save 3 lines of code, I think it's worth.